src/CacheManager.Core/Internal/BaseCache.cs

```diff
@@ -407,6 +410,55 @@ public virtual bool Remove(string key, string region)
         }
 
         /// <summary>
+        /// Gets all keys that match the given pattern.
+        /// </summary>
+        /// <param name="pattern">A glob to match against keys.</param>
+        /// <param name="region">The cache region.</param>
+        /// <returns>The matching unique keys in the cache</returns>
+        public virtual IEnumerable<string> Keys(string pattern, string region)
+        {
+            var keys = AllKeys();
+            if (region != null)
+                keys = keys.Where(k => k.StartsWith(region + ":"));
+
+            if (pattern == "*")
+            {
+                return keys;
+            }
+            if (pattern.Contains("*") || pattern.Contains("?"))
+            {
+                var regexPattern = "^" + Regex.Escape(pattern).Replace(@"\*", ".*").Replace(@"\?", ".") + "$";
```

Comment:
I was passing a glob, as that's what Redis uses for the server-side filter in SCAN, so I was keeping that as the pattern type to pass.

There's already shortcuts for the in-memory search (no wildcard, only wildcard), it would be easy (and I don't think very expensive) to look for the StartsWith/EndsWith cases and only fall back to regex for the complex cases.

As the performance of the matching could be both cache type and pattern dependent, and getting the key list is going to be relatively expensive anyway, maybe documenting the costs and advising profiling based on the users cases would be appropriate?